src/dht/dht_server.cc

```diff
@@ -597,7 +597,7 @@ void
 DhtServer::create_error(const DhtMessage& req, const rak::socket_address* sa, int num, const char* msg) {
   DhtMessage error;
 
-  if (req[key_t].is_raw_bencode() || req[key_t].is_raw_string())
+  if ((req[key_t].is_raw_bencode() || req[key_t].is_raw_string()) && req[key_t].as_raw_string().size < 20)
```

Comment:
The parenthesis placement is wrong, you can't call as_raw_string on is_raw_bencode.